Random failures in CI for gcc-10, Debug, ON

I've noticed this twice now on two separate CI runs in two separate PRs. 
1. https://github.com/sxs-collaboration/spectre/actions/runs/1356323620
```
make[2]: *** [CMakeFiles/Makefile2:20256: tests/Unit/DataStructures/Tensor/Expressions/CMakeFiles/Test_Expressions.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
[ 13%] Linking CXX shared library ../../../../../lib/libGeneralRelativityHelpers.so
[ 13%] Built target GeneralRelativityHelpers
make[1]: *** [CMakeFiles/Makefile2:9472: CMakeFiles/unit-tests.dir/rule] Error 2
make: *** [Makefile:161: unit-tests] Error 2
Error: Process completed with exit code 2.
```
2. https://github.com/sxs-collaboration/spectre/actions/runs/1365248969
```
make[2]: *** [CMakeFiles/Makefile2:20317: tests/Unit/DataStructures/Tensor/Expressions/CMakeFiles/Test_Expressions.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
[ 13%] Building CXX object tests/Unit/PointwiseFunctions/GeneralRelativity/CMakeFiles/Test_GeneralRelativity.dir/Test_ComputeSpacetimeQuantities.cpp.o
[ 13%] Building CXX object tests/Unit/PointwiseFunctions/GeneralRelativity/CMakeFiles/Test_GeneralRelativity.dir/Test_IndexManipulation.cpp.o
[ 13%] Building CXX object tests/Unit/PointwiseFunctions/GeneralRelativity/CMakeFiles/Test_GeneralRelativity.dir/Test_InterfaceNullNormal.cpp.o
[ 13%] Building CXX object tests/Unit/PointwiseFunctions/GeneralRelativity/CMakeFiles/Test_GeneralRelativity.dir/Test_KerrSchildCoords.cpp.o
[ 13%] Building CXX object tests/Unit/PointwiseFunctions/GeneralRelativity/CMakeFiles/Test_GeneralRelativity.dir/Test_ProjectionOperators.cpp.o
[ 13%] Building CXX object tests/Unit/PointwiseFunctions/GeneralRelativity/CMakeFiles/Test_GeneralRelativity.dir/Test_Ricci.cpp.o
[ 13%] Building CXX object tests/Unit/PointwiseFunctions/GeneralRelativity/CMakeFiles/Test_GeneralRelativity.dir/Test_Tags.cpp.o
[ 13%] Building CXX object tests/Unit/PointwiseFunctions/GeneralRelativity/CMakeFiles/Test_GeneralRelativity.dir/Test_Transform.cpp.o
[ 13%] Building CXX object tests/Unit/PointwiseFunctions/GeneralRelativity/CMakeFiles/Test_GeneralRelativity.dir/Test_WeylElectric.cpp.o
[ 13%] Building CXX object tests/Unit/PointwiseFunctions/GeneralRelativity/CMakeFiles/Test_GeneralRelativity.dir/Test_WeylPropagating.cpp.o
[ 13%] Linking CXX shared library ../../../../lib/libTest_GeneralRelativity.so
[ 13%] Built target Test_GeneralRelativity
make[1]: *** [CMakeFiles/Makefile2:9531: CMakeFiles/unit-tests.dir/rule] Error 2
make: *** [Makefile:161: unit-tests] Error 2
Error: Process completed with exit code 2.
```

Both failures appear to start with `make[2]: *** [CMakeFiles/Makefile2:20317: tests/Unit/DataStructures/Tensor/Expressions/CMakeFiles/Test_Expressions.dir/all] Error 2`. Unsure if that's a coincidence or not.

--- a/tests/Unit/DataStructures/Tensor/Expressions/CMakeLists.txt
+++ b/tests/Unit/DataStructures/Tensor/Expressions/CMakeLists.txt
@@ -9,7 +9,8 @@ set(LIBRARY_SOURCES
   Test_Contract.cpp
   Test_Divide.cpp
   Test_Evaluate.cpp
-  Test_EvaluateRank3.cpp
+  Test_EvaluateRank3NonSymmetric.cpp
+  Test_EvaluateRank3Symmetric.cpp
   Test_EvaluateRank4.cpp
   Test_EvaluateSpatialSpacetimeIndex.cpp
   Test_EvaluateTimeIndex.cpp

--- a/tests/Unit/DataStructures/Tensor/Expressions/Test_EvaluateRank3NonSymmetric.cpp
+++ b/tests/Unit/DataStructures/Tensor/Expressions/Test_EvaluateRank3NonSymmetric.cpp
@@ -1,0 +1,27 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+// Rank 3 test cases for TensorExpressions::evaluate are split into this file
+// and Test_EvaluateRank3Symmetric.cpp in order to reduce compile time memory
+// usage per cpp file.
+
+#include "Framework/TestingFramework.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Expressions/TensorIndex.hpp"
+#include "DataStructures/Tensor/IndexType.hpp"
+#include "Helpers/DataStructures/Tensor/Expressions/EvaluateRank3.hpp"
+
+SPECTRE_TEST_CASE(
+    "Unit.DataStructures.Tensor.Expression.EvaluateRank3NonSymmetric",
+    "[DataStructures][Unit]") {
+  // Rank 3: double; nonsymmetric
+  TestHelpers::TensorExpressions::test_evaluate_rank_3_no_symmetry<
+      double, SpacetimeIndex, SpatialIndex, SpacetimeIndex, UpLo::Up, UpLo::Lo,
+      UpLo::Up, ti_D, ti_j, ti_B>();
+
+  // Rank 3: DataVector; nonsymmetric
+  TestHelpers::TensorExpressions::test_evaluate_rank_3_no_symmetry<
+      DataVector, SpacetimeIndex, SpatialIndex, SpacetimeIndex, UpLo::Up,
+      UpLo::Lo, UpLo::Up, ti_D, ti_j, ti_B>();
+}

--- a/tests/Unit/DataStructures/Tensor/Expressions/Test_EvaluateRank3Symmetric.cpp
+++ b/tests/Unit/DataStructures/Tensor/Expressions/Test_EvaluateRank3Symmetric.cpp
@@ -1,6 +1,10 @@
 // Distributed under the MIT License.
 // See LICENSE.txt for details.
 
+// Rank 3 test cases for TensorExpressions::evaluate are split into this file
+// and Test_EvaluateRank3NonSymmetric.cpp in order to reduce compile time memory
+// usage per cpp file.
+
 #include "Framework/TestingFramework.hpp"
 
 #include "DataStructures/DataVector.hpp"
@@ -8,13 +12,9 @@
 #include "DataStructures/Tensor/IndexType.hpp"
 #include "Helpers/DataStructures/Tensor/Expressions/EvaluateRank3.hpp"
 
-SPECTRE_TEST_CASE("Unit.DataStructures.Tensor.Expression.EvaluateRank3",
-                  "[DataStructures][Unit]") {
-  // Rank 3: double; nonsymmetric
-  TestHelpers::TensorExpressions::test_evaluate_rank_3_no_symmetry<
-      double, SpacetimeIndex, SpatialIndex, SpacetimeIndex, UpLo::Up, UpLo::Lo,
-      UpLo::Up, ti_D, ti_j, ti_B>();
-
+SPECTRE_TEST_CASE(
+    "Unit.DataStructures.Tensor.Expression.EvaluateRank3Symmetric",
+    "[DataStructures][Unit]") {
   // Rank 3: double; first and second indices symmetric
   TestHelpers::TensorExpressions::test_evaluate_rank_3_ab_symmetry<
       double, SpacetimeIndex, SpacetimeIndex, UpLo::Lo, UpLo::Up, ti_b, ti_a,
@@ -33,11 +33,6 @@ SPECTRE_TEST_CASE("Unit.DataStructures.Tensor.Expression.EvaluateRank3",
   // Rank 3: double; symmetric
   TestHelpers::TensorExpressions::test_evaluate_rank_3_abc_symmetry<
       double, SpacetimeIndex, UpLo::Lo, ti_f, ti_d, ti_a>();
-
-  // Rank 3: DataVector; nonsymmetric
-  TestHelpers::TensorExpressions::test_evaluate_rank_3_no_symmetry<
-      DataVector, SpacetimeIndex, SpatialIndex, SpacetimeIndex, UpLo::Up,
-      UpLo::Lo, UpLo::Up, ti_D, ti_j, ti_B>();
 
   // Rank 3: DataVector; first and second indices symmetric
   TestHelpers::TensorExpressions::test_evaluate_rank_3_ab_symmetry<

